File: go/binary-tree-level-order-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 type TreeNodeLevel struct {
    Node *TreeNode
    Level int
}

func levelOrder(root *TreeNode) [][]int {
    s := []TreeNodeLevel{}
    s = append(s, TreeNodeLevel{Node: root, Level: 0})
    LevelOrder := [][]int{}
    for 0 < len(s) {
        treeNodeLevel := s[0]
        s = s[1:]
        node := treeNodeLevel.Node
        level := treeNodeLevel.Level
        if node == nil {
            continue
        }
        if level == len(LevelOrder) {
            LevelOrder = append(LevelOrder, []int{})
        }
        LevelOrder[level] = append(LevelOrder[level], node.Val)
        s = append(s, TreeNodeLevel{Node: node.Left, Level: level + 1})
        s = append(s, TreeNodeLevel{Node: node.Right, Level: level + 1})
    }
    return LevelOrder
}

File: python/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        level_order = []
        s = [(root, 0)]
        while s:
            node, level = s.pop()
            if node == None:
                continue
            if level == len(level_order):
                level_order.append([])
            level_order[level].append(node.val)
            s.append((node.right, level + 1))
            s.append((node.left, level + 1))
        return level_order
